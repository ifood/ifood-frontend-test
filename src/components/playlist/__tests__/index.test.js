import React from 'react'
import { useSelector } from 'react-redux'
import '@testing-library/jest-dom'
import { render, cleanup } from '@testing-library/react'

import { PlaylistProvider, PlaylistElement } from '../index'

afterEach(cleanup)

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => jest.fn(),
}))

describe('<Playlist />', () => {
  beforeEach(() => {
    useSelector.mockImplementation((cb) =>
      cb({
        filter: {
          filters: [],
          currentFilters: {
            name: 'mocked',
          },
          hide: true,
        },
        playlist: {
          playlists: [
            { name: 'mocked playlist' },
            { name: 'not mocked playlist 2' },
          ],
          loading: false,
          error: null,
          total: 20,
        },
      })
    )
  })

  afterEach(() => {
    useSelector.mockClear()
  })

  it('Should be able to render without errors', () => {
    const { getByText } = render(<PlaylistProvider />)

    expect(getByText('mocked playlist')).toBeTruthy()
    expect(getByText('not mocked playlist 2')).toBeTruthy()
  })

  it('Snapshot testing Playlist', () => {
    const mockedName = 'mocked mocked'
    const mockedDescription = '<a>Mocked description</a>'
    const { asFragment } = render(
      <PlaylistElement name={mockedName} description={mockedDescription} />
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('Should be able to render moked props', async () => {
    const mockedName = 'mocked name'
    const mockedDescription = 'Mocked description'
    // eslint-disable-next-line prettier/prettier
    const { getByText, queryByTestId } = render(<PlaylistElement name={mockedName} description={mockedDescription} />)

    const playlistContainer = queryByTestId('playlist-container')

    expect(playlistContainer).not.toBeNull()
    expect(getByText('mocked name')).toBeTruthy()
  })
})
