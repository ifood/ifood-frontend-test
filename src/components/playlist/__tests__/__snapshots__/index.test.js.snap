// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Playlist /> Snapshot testing Playlist 1`] = `
<DocumentFragment>
  <div
    class="sc-dlnjPT gedplC"
    data-testid="playlist-container"
  >
    <img
      alt="mocked mocked"
      class="sc-hKFyIo jJzsPK"
      src="/"
    />
    <h1
      class="sc-eCApGN dxliyU"
    >
      mocked mocked
    </h1>
    <h2
      class="sc-jSFkmK fMtTlC"
    >
      <a>
        Mocked description
      </a>
    </h2>
  </div>
</DocumentFragment>
`;
